### ISSUE TITLE
fix: ✨ sd-button a11y rem issue

### ISSUE THREAD

**Issue by karlbaumhauer** [solid-design-system/solid#1546]
## User Story
We need to discuss how we want to handle rem vs. px issues as the topic came up again in our a11y audit => https://www.notion.so/sd-badge-103f3f6e1bc080619981e3e8d9546a9f?d=7e85df0b8e43468297c5a7c89b1e0b01&pvs=4#104f3f6e1bc080b0be0ec6392d47b458

### Suggested Solution
either switch to rem or use a tool to convert to rem?!?

Fix button (line-height?) issues as stated in the comments.
`sd-badge` problem has been fixed already.

## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

**Comment by mariohamann** [comment]
The audit explicitly stated this for:
- the `sd-badge`
- for `font-size`

When looking deeper into the code, it's obvious why: The font-size of the `sd-badge` was written as an arbitrary value `text-[12px]`. All other font-sizes are automatically set with `rem`.

Therefore this could/should be changed in the `sd-badge` ticket directly @smfonseca. Maybe we find other occasions as well. Nevertheless while increasing the default text size in the browser settings I saw some other issues:

Firefox:
![CleanShot 2024-10-16 at 07 06 41@2x](https://github.com/user-attachments/assets/ce2e2511-bdba-4ba4-b97b-b17a2e05c8b3)

Chromium:
![CleanShot 2024-10-16 at 07 10 33@2x](https://github.com/user-attachments/assets/388539ab-764b-449b-b270-3be2f83ead00)

Safari (Bug in Button):
![CleanShot 2024-10-16 at 07 11 04@2x](https://github.com/user-attachments/assets/8a83cffe-8fb8-4ab4-b465-4802ac441a89)

**Comment by yoezlem** [comment]
The ticket needs to be rewritten – we only have this issue with the badge and the button.
